### PR TITLE
test: Improve the test to check status

### PR DESCRIPTION
**Pull request by zhuchenwang**
Also add the nfs server to the test setup.

Signed-off-by: Zhuchen Wang <zcwang@google.com>

### CODE REVIEW

File: examples/vm_disk_example/main.tf [hunk 1]
```diff
@@ -24,10 +24,11 @@ module "emtpy_disk" {
   disk_size = "10Gi"
 }
 
+# Use a small OS image here because downloading from HTTP is very slow in the test setup.
 module "disk_from_http" {
   source = "../../modules/vm-disk"
   http_source = {
-    url = "https://cloud-images.ubuntu.com/releases/focal/release/ubuntu-20.04-server-cloudimg-amd64.img"
+    url = "https://download.cirros-cloud.net/0.6.0/cirros-0.6.0-x86_64-disk.img"
```
Comment (apeabody) [inline examples/vm_disk_example/main.tf:31]: If you want a "production" OS that is also very small, you might want to consider https://hub.docker.com/_/alpine

File: test/setup/scripts/install_hybrid_cluster.sh [hunk 2]
```diff
@@ -149,11 +149,15 @@ export GOOGLE_APPLICATION_CREDENTIALS=/root/bm-gcr.json
...
 gcloud compute scp --zone "${ZONE}" "${CLUSTER_YAML_FILE}" root@"${VM_WS}":/root/bmctl-workspace/"${CLUSTER_ID}"/"${CLUSTER_ID}".yaml
+gcloud compute scp --zone "${ZONE}" "${NFS_YAML_FILE}" root@"${VM_WS}":/root/nfs-csi.yaml
 gcloud compute ssh root@"${VM_WS}" --zone "${ZONE}" <<EOF
 set -x
 export GOOGLE_APPLICATION_CREDENTIALS=/root/bm-gcr.json
+export KUBECONFIG=/root/bmctl-workspace/${CLUSTER_ID}/${CLUSTER_ID}-kubeconfig
 bmctl create cluster -c ${CLUSTER_ID}
-bmctl enable vmruntime --kubeconfig=/root/bmctl-workspace/${CLUSTER_ID}/${CLUSTER_ID}-kubeconfig
+bmctl enable vmruntime --kubeconfig=\${KUBECONFIG}
+bash -c 'curl -skSL https://raw.githubusercontent.com/kubernetes-csi/csi-driver-nfs/v3.1.0/deploy/install-driver.sh | bash -s v3.1.0 --'
```
[the remark this answers is not in the document]
Comment (zhuchenwang) [reply test/setup/scripts/install_hybrid_cluster.sh:159]: Actually I tried to use the v4.1.0 but the filestore server only supports NFS v3. https://cloud.google.com/filestore/docs/overview#protocol_support

[the remark this answers is not in the document]
Comment (apeabody) [reply test/setup/scripts/install_hybrid_cluster.sh:159]: Hmm, I don't believe that the nfs-csi driver version is coordinated to the NFS server version, in fact we specify the NFS server version here: https://github.com/GoogleCloudPlatform/terraform-google-anthos-vm/pull/8/files#diff-63865bdff3759fb57a698e58ef4709910b18d69642b7fbee53bccd2935c73a8eR38

The v3.1 driver is likely fine for now, but do you recall the issue with v4.1?

[the remark this answers is not in the document]
Comment (zhuchenwang) [reply test/setup/scripts/install_hybrid_cluster.sh:159]: After updating to v4.1, I got the error `mount.nfs: Protocol not supported`. Then I realize that I need to change the mount parameter. However, I got the error `mount.nfs: access denied by server while mounting` after updating the `nfsvers=3`. Then I deleted the v4.1 driver and install the v3.1 driver. Everything started to work.

[the remark this answers is not in the document]
Comment (zhuchenwang) [reply test/setup/scripts/install_hybrid_cluster.sh:159]: Let me try it again locally.

[the remark this answers is not in the document]
Comment (zhuchenwang) [reply test/setup/scripts/install_hybrid_cluster.sh:159]: Here is the error on the client side.
```
Mounting command: mount
Mounting arguments: -t nfs -o hard,timeo=600,retrans=3,rsize=1048576,wsize=1048576,resvport,nfsvers=3 10.244.107.82:cluster1_fs /tmp/pvc-46682d4a-95c4-4c64-9d14-64d5dadf78e1
Output: mount.nfs: access denied by server while mounting {IP}:cluster1_fs
```

[the remark this answers is not in the document]
Comment (apeabody) [reply test/setup/scripts/install_hybrid_cluster.sh:159]: I opened #10 to track

**Pull request merged**